### PR TITLE
Fix changing of penalty codes

### LINKED ISSUE #420: Custom penalties not being picked up
When you define a new ruleset pointing to a custom penalty file and then use it, the standard wftda 2018 penalties are being used still.

### PR DESCRIPTION
**Pull request by brian-brazil**
Fix string comparison

Use standard way of storing data, rather than hand implementing it.

Fixes #420

### CODE REVIEW

File: src/com/carolinarollergirls/scoreboard/penalties/PenaltyCodesManager.java [hunk 1]
```diff
@@ -23,31 +23,14 @@
...
     }
 
-    public ValueWithId _get(AddRemoveProperty prop, String id, boolean add) {
-        for (PenaltyCode c : getDefinitions()) {
-            if (c.getId().equals(id)) { return c; }
-        }
-        return null;
-    }
-    @Override
-    public Collection<? extends ValueWithId> getAll(AddRemoveProperty prop) { return getDefinitions(); }
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/penalties/PenaltyCodesManager.java:37]: Collection and List are now unused imports.

File: src/com/carolinarollergirls/scoreboard/penalties/PenaltyCodesManager.java [hunk 2]
```diff
@@ -23,31 +23,14 @@
     public PenaltyCodesManager(ScoreBoard parent) {
         super(parent, null, "", ScoreBoard.Child.PENALTY_CODES, PenaltyCodesManager.class, Child.class);
         this.parent = parent;
-        definitions = loadFromJSON(parent.getRulesets().get(Rule.PENALTIES_FILE));
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/penalties/PenaltyCodesManager.java:26]: We should still initialize the definitions.

File: src/com/carolinarollergirls/scoreboard/penalties/PenaltyCodesManager.java [hunk 3]
```diff
@@ -60,13 +43,11 @@ public PenaltyCodesDefinition loadFromJSON(String file) {
...
     }
 
-    private PenaltyCodesDefinition definitions;
-
     protected ScoreBoardListener rulesetChangeListener = new ScoreBoardListener() {
         @Override
         public void scoreBoardChange(ScoreBoardEvent event) {
             ValueWithId v = (ValueWithId)event.getValue();
-            if (v.getId() == Rule.PENALTIES_FILE.toString()) {
+            if (v.getId().equals(Rule.PENALTIES_FILE.toString())) {
```
Comment (frank-weinberg) [inline src/com/carolinarollergirls/scoreboard/penalties/PenaltyCodesManager.java]: `v` might be null, it's probably better to compare the other way.

**Comment by brian-brazil** [comment]
All fixed.

**Pull request merged**